pysipp/launch.py

```diff
@@ -21,116 +22,109 @@ class TimeoutError(Exception):
     "SIPp process timeout exception"
 
 
-class PopenRunner(object):
-    """Run a sequence of SIPp agents asynchronously. If any process terminates
-    with a non-zero exit code, immediately kill all remaining processes and
-    collect std streams.
+class SIPpFailure(RuntimeError):
+    """SIPp commands failed
+    """
+
 
-    Adheres to an interface similar to `multiprocessing.pool.AsyncResult`.
+class TrioRunner(object):
+    """Run a sequence of SIPp cmds asynchronously. If any process terminates
+    with a non-zero exit code, immediately canacel all remaining processes and
+    collect std streams.
     """
     def __init__(
         self,
-        subprocmod=subprocess,
-        osmod=os,
-        poller=select.epoll,
     ):
-        # these could optionally be rpyc proxy objs
-        self.spm = subprocmod
-        self.osm = osmod
-        self.poller = poller()
-        # collector thread placeholder
-        self._waiter = None
         # store proc results
         self._procs = OrderedDict()
 
-    def __call__(self, cmds, block=True, rate=300, **kwargs):
-        if self._waiter and self._waiter.is_alive():
+    async def run(
+        self,
+        cmds,
+        rate=300,
+        **kwargs
+    ):
+        if self.is_alive():
             raise RuntimeError(
                 "Not all processes from a prior run have completed"
             )
         if self._procs:
             raise RuntimeError(
                 "Process results have not been cleared from previous run"
             )
-        sp = self.spm
-        os = self.osm
-        DEVNULL = open(os.devnull, 'wb')
-        fds2procs = OrderedDict()
-
         # run agent commands in sequence
         for cmd in cmds:
             log.debug(
                 "launching cmd:\n\"{}\"\n".format(cmd))
-            proc = sp.Popen(
+            # proc = await trio.open_process(
+            proc = trio.Process(
```

Comment:
This api is now old.

We should probably port to the newer [`trio.open_process()`](https://trio.readthedocs.io/en/stable/reference-io.html#spawning-subprocesses).